velox/expression/ExprToSubfieldFilter.h

```diff
@@ -363,7 +370,8 @@ betweenHugeint(int128_t min, int128_t max, bool nullAllowed = false) {
 
 std::pair<common::Subfield, std::unique_ptr<common::Filter>> toSubfieldFilter(
     const core::TypedExprPtr& expr,
-    core::ExpressionEvaluator*);
+    core::ExpressionEvaluator* evaluator,
+    bool negated = false);
```

Comment:
It looks like this PR may combine multiple changes. Some might be optimizations that do not change behavior, others are API changes. Would it be possible to split these into separate PRs and add tests for newly introduced functionality?

Reply:
This just add recursive case, it can be implemented as separate recursive function in anonymous namespace, but why? So I think api wasn't changed

Reply:
> This function is exposed in the header file, hence, I assume it is public API. It used to have 2 parameters, now it has 3. This seems like a change.

It's has default value, so it's not changed, it's extended.

> I see this API wasn't documented before. Would you help document it?

ok